source/vir/src/interpreter.rs

```diff
@@ -0,0 +1,1438 @@
+                                    let call = exp_new(CallLambda(
+                                        lambda.typ.clone(),
+                                        lambda.clone(),
+                                        args,
+                                    ));
+                                    eval_expr_internal(ctx, state, &call)
+                                })
+                                .collect();
+                            let im_vec: Vector<Exp> = vec?.into_iter().collect();
+                            seq_new(im_vec)
+                        }
+                        _ => ok,
+                    }
+                }
+                Push => match &args[0].x {
+                    Interp(Seq(s)) => {
+                        let mut s = s.clone();
+                        s.push_back(args[1].clone());
+                        seq_new(s)
+                    }
+                    _ => ok,
+                },
+                Update => match &args[0].x {
+                    Interp(Seq(s)) => match get_int(&args[1]) {
+                        Some(index) if index < s.len() => {
+                            let s = s.update(index, args[2].clone());
+                            seq_new(s)
+                        }
+                        _ => ok,
+                    },
+                    _ => ok,
+                },
+                Subrange => match &args[0].x {
+                    Interp(Seq(s)) => {
+                        let start = get_int(&args[1]);
+                        let end = get_int(&args[2]);
+                        match (start, end) {
+                            (Some(start), Some(end)) if start <= end && end <= s.len() => {
+                                seq_new(s.clone().slice(start..end))
+                            }
+                            _ => ok,
+                        }
+                    }
+                    _ => ok,
+                },
+                Add => match (&args[0].x, &args[1].x) {
+                    (Interp(Seq(s1)), Interp(Seq(s2))) => {
+                        let mut s = s1.clone();
+                        s.append(s2.clone());
+                        seq_new(s)
+                    }
+                    _ => ok,
+                },
+                Len => match &args[0].x {
+                    Interp(Seq(s)) => int_new(BigInt::from_usize(s.len()).unwrap()),
+                    _ => ok,
+                },
+                Index => match &args[0].x {
+                    Interp(Seq(s)) => match &args[1].x {
+                        UnaryOpr(crate::ast::UnaryOpr::Box(_), e) => match &e.x {
+                            Const(Constant::Int(index)) => {
+                                match BigInt::to_usize(index) {
+                                    None => {
+                                        // TODO: Use Diagnostics instead of println
+                                        println!(
```

Comment:
It looks like diagnostics are important for these warnings (so that we can emit a span). `well_formed` collects VIR-generated warnings in a Vec https://github.com/secure-foundations/verus/blob/main/source/vir/src/well_formed.rs#L780 and lets `rust_verify` handle the printing. We may want to do something similar, but it may be a bit trickier for this, because you'd have to thread this through the contexts, I think.

Reply:
Indeed, I suspect it might be awkward to pass the warnings all the way back up.  At some point, you, Chris, and I had talked about defining a Diagnostics-like trait in AIR, which rust_verify could implement using its own Diagnostics.  That would let us clean up the `println`s in the profiler (e.g., https://github.com/secure-foundations/verus/blob/e72629774e0643370ff9188bb4e8920ef23d9ab6/source/air/src/profiler.rs#L39), and if we're already passing it through VIR to AIR, then it would be natural to use here too.  What do you think?

Reply:
That sounds good!

Reply:
Cool, as discussed in yesterday's meeting, I'll plan on adding that after this lands.